File: examples/java/kora-java-crud-submodule/kora-java-crud-submodule-pet-api/src/main/java/io/koraframework/example/submodule/pet/PetModule.java
package io.koraframework.example.submodule.pet;

import io.koraframework.cache.caffeine.CaffeineCacheModule;
import io.koraframework.common.annotation.KoraSubmodule;
import io.koraframework.database.jdbc.JdbcDatabaseModule;
import io.koraframework.example.submodule.common.CommonModule;
import io.koraframework.resilient.ResilientModule;

@KoraSubmodule
public interface PetModule extends
        CommonModule,
        JdbcDatabaseModule,
        CaffeineCacheModule,
        ResilientModule {

}

File: examples/kotlin/kora-kotlin-crud-submodule/kora-kotlin-crud-submodule-pet-api/src/main/kotlin/io/koraframework/kotlin/example/submodule/pet/PetModule.kt
package io.koraframework.kotlin.example.submodule.pet

import io.koraframework.cache.caffeine.CaffeineCacheModule
import io.koraframework.common.annotation.KoraSubmodule
import io.koraframework.database.jdbc.JdbcDatabaseModule
import io.koraframework.kotlin.example.submodule.common.CommonModule
import io.koraframework.resilient.ResilientModule

@KoraSubmodule
interface PetModule : CommonModule, JdbcDatabaseModule, CaffeineCacheModule, ResilientModule
